pkg/operator/certrotation/cabundle.go

```diff
@@ -64,10 +64,11 @@ func (c CABundleConfigMap) EnsureConfigMapCABundle(ctx context.Context, signingC
 	}
 	needsMetadataUpdate = c.AdditionalAnnotations.EnsureTLSMetadataUpdate(&caBundleConfigMap.ObjectMeta) || needsMetadataUpdate
 	if needsMetadataUpdate && len(caBundleConfigMap.ResourceVersion) > 0 {
-		_, _, err := resourceapply.ApplyConfigMap(ctx, c.Client, c.EventRecorder, caBundleConfigMap)
+		actualCABundleConfigMap, _, err := resourceapply.ApplyConfigMap(ctx, c.Client, c.EventRecorder, caBundleConfigMap)
```

Comment:
do you mind adding a unit test ?

Reply:
Done, not sure if it really ensures hotlooping is not happening (imo fuzzing would confirm it)

Reply:
This PR doesn't solve the hotloop issue. The hotloop issue was resolved by https://github.com/openshift/cluster-kube-apiserver-operator/pull/1661.

There were `4` controllers fighting over `kube-control-plane-signer-ca`

Reply:
Fixed this by allowing metadata change to the first controller only (whoever has set the first ownership ref).

I'll run this a few time to make sure this won't make TLS metadata jitter from run to run